x-pack/plugins/alerting/server/rules_client/common/snooze_utils.ts

```diff
@@ -117,7 +117,7 @@ export function clearUnscheduledSnooze<Params extends RuleParams>(rule: RuleDoma
 
 export function clearScheduledSnoozesAttributesById(attributes: RuleAttributes, ids: string[]) {
   return attributes.snoozeSchedule
-    ? attributes.snoozeSchedule.filter((s) => s.id && !ids.includes(s.id))
+    ? attributes.snoozeSchedule.filter((s) => !s.id || !ids.includes(s.id))
```

Comment:
nitpick: I find it easier to reason about if its: `!(s.id && ids.include(s.id))`